core-bundle/tests/Controller/AbstractBackendControllerTest.php

```diff
@@ -0,0 +1,122 @@
+            }
+        };
+
+        // Legacy setup
+        \Contao\Environment::reset();
+        (new Filesystem())->mkdir($this->getTempDir().'/languages/en');
+
+        $GLOBALS['TL_LANG']['MSC'] = [
+            'version' => 'version',
+            'dashboard' => 'dashboard',
+            'home' => 'home',
+            'learnMore' => 'learn more',
+        ];
+
+        $GLOBALS['TL_LANGUAGE'] = 'en';
+
+        $_SERVER['HTTP_USER_AGENT'] = 'Contao/Foo';
+        $_SERVER['HTTP_HOST'] = 'localhost';
+
+        if (!\defined('TL_FILES_URL')) {
+            \define('TL_FILES_URL', '');
+        }
+
+        $expectedContext = [
+            'version' => 'my version',
+            'headline' => 'dashboard',
+            'title' => '',
+            'theme' => 'flexible',
+            'base' => 'http://localhost/',
+            'language' => 'en',
+            'host' => 'localhost',
+            'charset' => 'UTF-8',
+            'home' => 'home',
+            'isPopup' => null,
+            'learnMore' => 'learn more',
+            'menu' => '<menu>',
+            'headerMenu' => '<header_menu>',
+            'foo' => 'bar',
+        ];
+
+        $container = $this->getContainerWithDefaultConfiguration($expectedContext);
+
+        System::setContainer($container);
+        $controller->setContainer($container);
+
+        $this->assertSame('<custom_be_main>', $controller->fooAction()->getContent());
+
+        // Cleanup
+        unset($GLOBALS['TL_LANG'], $GLOBALS['TL_LANGUAGE'], $_SERVER['HTTP_USER_AGENT'], $_SERVER['HTTP_HOST']);
+    }
+
+    private function getContainerWithDefaultConfiguration(array $expectedContext): ContainerBuilder
+    {
+        $container = $this->getContainerWithContaoConfiguration($this->getTempDir());
+
+        $authorizationChecker = $this->createMock(AuthorizationCheckerInterface::class);
+        $authorizationChecker
+            ->method('isGranted')
+            ->with('ROLE_USER')
+            ->willReturn(true)
+        ;
+
+        $twig = $this->createMock(Environment::class);
+        $twig
+            ->expects($this->exactly(3))
+            ->method('render')
+            ->willReturnMap([
+                ['@ContaoCore/Backend/be_menu.html.twig', [], '<menu>'],
+                ['@ContaoCore/Backend/be_header_menu.html.twig', [], '<header_menu>'],
+                ['custom_be.html.twig', $expectedContext, '<custom_be_main>'],
+            ])
+        ;
+
+        $container->set('security.authorization_checker', $authorizationChecker);
+        $container->set('security.token_storage', $this->createMock(TokenStorageInterface::class));
+        $container->set('contao.security.token_checker', $this->createMock(TokenChecker::class));
+        $container->set('database_connection', $this->createMock(Connection::class));
+        $container->set('session', $this->createMock(Session::class));
```

Comment:
Why are you mocking the deprecated `Contao\Session` class instead of the Symfony session?

Reply:
Good catch, that was just a mistake taking the wrong namespace. Fixed in cb50a0a417eda659af367ffe25f1f75b401c76c8